lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/cache/TestEvictableCache.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.cache;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheStats;
+import io.airlift.concurrent.MoreFutures;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static java.lang.String.format;
+import static java.util.concurrent.Executors.newFixedThreadPool;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+public class TestEvictableCache
+{
+    private static final int TEST_TIMEOUT_MILLIS = 10_000;
+
+    private enum Invalidation
+    {
+        INVALIDATE_KEY,
+        INVALIDATE_PREDEFINED_KEYS,
+        INVALIDATE_SELECTED_KEYS,
+        INVALIDATE_ALL,
+    }
+
+    @Test(timeOut = TEST_TIMEOUT_MILLIS)
+    public void testLoad()
+            throws Exception
+    {
+        Cache<Integer, String> cache = EvictableCache.buildWith(
+                CacheBuilder.newBuilder());
+
+        assertEquals(cache.stats(), new CacheStats(0, 0, 0, 0, 0, 0));
+        String value = cache.get(42, () -> "abc");
+        assertEquals(cache.stats(), new CacheStats(0, 1, 1, 0, cache.stats().totalLoadTime(), 0));
+        assertEquals(value, "abc");
+    }
+
+    /**
+     * Covers https://github.com/google/guava/issues/1881
+     */
+    @Test(timeOut = TEST_TIMEOUT_MILLIS, dataProvider = "invalidations")
+    public void testInvalidateOngoingLoad(Invalidation invalidation)
+            throws Exception
+    {
+        Cache<Integer, String> cache = EvictableCache.buildWith(CacheBuilder.newBuilder());
```

Comment:
Add 

```java
    public static <K, V> EvictableCache<K, V> build()
    {
        return new EvictableCache<>(CacheBuilder.newBuilder());
    }
```

?

Reply:
That's not a useful cache, as it would be unbounded, so i won't add it in `EvictableCache`.

In the test itself, it doesn't matter.